### PR TITLE
fix(Angular): properly compare RegExp with other objects for equality

### LINKED ISSUE #11204: angular.equals(/lol/, new Date)  // TRUE
angular.equals(/lol/, new Date)  // TRUE 

bug

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #11204

### CODE REVIEW

**Comment by Maxmaxmaximus** [comment]
angular.equals(/lol/, new Date) // TRUE

**Comment by Maxmaxmaximus** [comment]
angular.equals(/lol/, new Date) // TRUE

File: src/Angular.js [hunk 1]
```diff
@@ -860,8 +860,8 @@ function equals(o1, o2) {
       } else if (isDate(o1)) {
         if (!isDate(o2)) return false;
         return equals(o1.getTime(), o2.getTime());
-      } else if (isRegExp(o1) && isRegExp(o2)) {
-        return o1.toString() == o2.toString();
+      } else if (isRegExp(o1)) {
+        return isRegExp(o2) ? o1.toString() == o2.toString() : false;
       } else {
         if (isScope(o1) || isScope(o2) || isWindow(o1) || isWindow(o2) || isArray(o2)) return false;
```
Comment (alexlamsl) [inline src/Angular.js]: Not the same but a similar bug:
```
angular.equals({}, /lol/);
```

Comment (pkozlowski-opensource) [inline src/Angular.js]: @alexlamsl you are right - going to push an update for those cases as well

**Comment by pkozlowski-opensource** [comment]
@petebacondarwin / @lgalfaso / @Narretz could one of you guys OK this patch? It is pretty minimal and should be fast to review. Thnx!

**Comment by petebacondarwin** [comment]
LGTM

**Pull request closed without merging**